experimental/Schemes/CartierDivisor.jl

```diff
@@ -0,0 +1,40 @@
+export CartierDivisor
+export trivializing_covering
+
+@attributes mutable struct CartierDivisor{
+                                          CoveredSchemeType<:AbsCoveredScheme
+                                         }
+    X::CoveredSchemeType
+    I::IdealSheaf
+    C::Covering
+
+    function CartierDivisor(
+            X::AbsCoveredScheme, 
+            D::IdDict{<:AbsSpec, <:RingElem};
+            check::Bool=true
+        )
+        ID = IdDict{AbsSpec, Ideal}()
+        for U in keys(D)
+            ID[U] = ideal(OO(U), D[U])
+        end
+        I = IdealSheaf(X, ID, check=check)
+        if check
+          for U in keys(D)
+            is_zero_divisor(OO(U)(D[U])) && error("local elements must not be zero divisors")
+          end
+          # TODO: 
+          # - Check that every affine chart is covered
+        end
+        C = Covering(collect(keys(D)), IdDict{Tuple{AbsSpec, AbsSpec}, AbsGlueing}())
+        return new{typeof(X)}(X, I, C)
+    end
+end
+
+function (C::CartierDivisor)(U::AbsSpec)
+    return gens(C.I(U))
+end
+
+variety(C::CartierDivisor) = C.X
```

Comment:
`X` is not necessarily a variety but rather an arbitrary scheme .  So this seems odd to me.

Reply:
Why? Cartier divisors can be defined on arbitrary schemes.

Reply:
Yes. This command is missleading if X is not a variety. E.g. not over an algebraically closed field, not irreducible or not reduced. Hence I prefer the name scheme.

Reply:
Yes, you are right. Sorry, I didn't get it at first.